### ISSUE TITLE
Magento 2 Bin directory

### ISSUE THREAD

**Issue by jamescowie** [magento/magento2#1429]
So magento now has a `bin` directory where the new and great `magento` command can be found. 
However it is a common configuration option for projects to also use `bin` for composer binaries that are available. E.g. `Behat, PHPSpec, PHPUnit` etc. 

Because in the package `magento2-base` there is the mapping information for:

```
            [
                "bin",
                "bin"
            ],
``` 

When the composer and the hackathon installer process run after an update / install then the project roots `bin` directory is cleared out and only the magento 2 directory is created. This causes problems for projects that are configured to run the binaries from `./bin`. Granted the binaries are still available from `./vendor/package/bin/package` however it would be ideal to remove the mapping from `magento2-base` and replace with the following configuration in composer.json

`"bin": ["bin/magento”]`

This would mean that composer can place the `magento` command into the projects configured bin directory offering a single configuration node for the `bin` folder that is managed via `composer` and `magento` 

What do you think ? 

This relates to MAGETWO-38518 and #1351 but contains more information on what is causing the replacement.

**Comment by buskamuza** [comment]
@jamescowie , we're currently working on this fix (MAGETWO-38518) and you described exactly what we're going to change. So, I believe, we should cover the case you describe.
The fix should be available soon, but depending on the publication to this repository and other internal procedures it may take couple weeks.

**Comment by vancoz** [comment]
Also looks like we have reported duplicate issue https://github.com/magento/magento2/issues/1351

**Comment by jamescowie** [comment]
Hi @vancoz yep this is a reported issue of the other ticket I raised. Happy to close this or the other. Reason for a new ticket was to provide more info on a single issue that was not git workflow :) 

While continuing my work on M2 and related to this issue one really annoying thing other than getting my bin dir wiped out each update is that it seems that the permissions for `bin/magento` are changed on each release of the beta. I found that the permissions are not executable when I clone down.. Is this intentional ?
